kartothek/serialization/_generic.py

```diff
@@ -239,6 +239,12 @@ def filter_array_like(
     elif is_list_like(value) and len(value) and isinstance(value[0], date_type_to_cast):
         value = [pd.Timestamp(val).to_datetime64() for val in value]
 
+    # NULL types might not be preserved well, so try to cast floats (pandas default type) to the value type
+    if np.issubdtype(array_like.dtype, np.floating) and np.isnan(array_like).all():
```

Comment:
Is this what was causing the warning of the numpy equals warning? Is this something we should add an explicit test for? I'm having issues estimating the impact of this

Reply:
The issue here are CSVs that have incorrect types when being recovered. So empty columns won't be `object` but `float`, which causes the warning.